src/structure/viewer.ts

```diff
@@ -1206,54 +1205,59 @@ export class MoleculeViewer {
 
         const active_shapes = this._options.shape.value.split(',');
 
-        // consolidates all shapes in a single CustomShapeSpec
-        const all_shapes: $3Dmol.CustomShapeSpec = {
+        // merge all shapes in a single CustomShapeSpec
+        const mergedShapes: $3Dmol.CustomShapeSpec = {
             vertexArr: [],
             normalArr: [],
             faceArr: [],
             color: [],
         };
 
         // color function for atoms
-        const colorfunc = this._colorFunction();
-        const atomspec: $3Dmol.AtomSpec = {};
+        const colorFunc = this._colorFunction();
+        const atomSpec: $3Dmol.AtomSpec = {};
 
         for (const shape of active_shapes) {
             if (shape === '') {
                 continue;
             }
+
             assert(shape in structure.shapes);
-            const current_shape = structure.shapes[shape];
-            const supercell_a = this._options.supercell[0].value;
-            const supercell_b = this._options.supercell[1].value;
-            const supercell_c = this._options.supercell[2].value;
+            const currentShape = structure.shapes[shape];
+            let supercell_a = this._options.supercell[0].value;
+            let supercell_b = this._options.supercell[1].value;
+            let supercell_c = this._options.supercell[2].value;
             let cell = this._current.structure.cell;
 
-            if ((supercell_a > 1 || supercell_b > 1 || supercell_c > 1) && cell === undefined) {
-                return;
```

Comment:
this was the main issue in https://github.com/lab-cosmo/chemiscope/issues/361#issuecomment-2325084876